assets/js/blocks/rating-filter/test/block.tsx

```diff
@@ -0,0 +1,497 @@
+	);
+
+	const getList = () => container.querySelector( selectors.list );
+	const getDropdown = () => screen.queryByRole( 'combobox' );
+
+	const getChipsContainers = () =>
+		container.querySelectorAll( selectors.chipsContainer );
+	const getSuggestionsContainer = () =>
+		container.querySelector( selectors.suggestionsContainer );
+
+	const getChips = ( value: string ) => {
+		const chipsContainers = getChipsContainers();
+		const chips = Array.from( chipsContainers ).find( ( chipsContainer ) =>
+			chipsContainer
+				? within( chipsContainer ).queryByLabelText(
+						`Rated ${ value } out of 5`
+				  )
+				: false
+		);
+
+		return chips || null;
+	};
+	const getSuggestion = ( value: string ) => {
+		const suggestionsContainer = getSuggestionsContainer();
+		if ( suggestionsContainer ) {
+			return within( suggestionsContainer ).queryByLabelText(
+				`Rated ${ value } out of 5`
+			);
+		}
+		return null;
+	};
+	const getCheckbox = ( value: string ) => {
+		const checkboxesContainer = getList();
+		const checkboxes = checkboxesContainer
+			? checkboxesContainer.querySelectorAll( 'input' )
+			: [];
+
+		const checkbox = Array.from( checkboxes ).find(
+			( input ) => input.id === value
+		);
+
+		return checkbox;
+	};
+
+	const getRemoveButtonFromChips = ( chips: HTMLElement | null ) =>
+		chips
+			? within( chips ).getByLabelText( 'Remove rating filter.' )
+			: null;
+
+	const getRating2Chips = () => getChips( '2' );
+	const getRating4Chips = () => getChips( '4' );
+	const getRating5Chips = () => getChips( '5' );
+
+	const getRating2Suggestion = () => getSuggestion( '2' );
+	const getRating4Suggestion = () => getSuggestion( '4' );
+	const getRating5Suggestion = () => getSuggestion( '5' );
+
+	const getRating2Checkbox = () => getCheckbox( '2' );
+	const getRating4Checkbox = () => getCheckbox( '4' );
+	const getRating5Checkbox = () => getCheckbox( '5' );
+
+	return {
+		...utils,
+		container,
+		getDropdown,
+		getList,
+		getRating2Chips,
+		getRating4Chips,
+		getRating5Chips,
+		getRating2Suggestion,
+		getRating4Suggestion,
+		getRating5Suggestion,
+		getRating2Checkbox,
+		getRating4Checkbox,
+		getRating5Checkbox,
+		getRemoveButtonFromChips,
+	};
+};
+
+interface SetupParams {
+	filterRating: string;
+	displayStyle: DisplayStyle;
+	selectType: SelectType;
+}
+
+const setupSingleChoiceList = ( filterRating = '5' ) =>
+	setup( {
+		filterRating,
+		displayStyle: 'list',
+		selectType: 'single',
+	} );
+
+const setupMultipleChoiceList = ( filterRating = '5' ) =>
+	setup( {
+		filterRating,
+		displayStyle: 'list',
+		selectType: 'multiple',
+	} );
+
+const setupSingleChoiceDropdown = ( filterRating = '5' ) =>
+	setup( {
+		filterRating,
+		displayStyle: 'dropdown',
+		selectType: 'single',
+	} );
+
+const setupMultipleChoiceDropdown = ( filterRating = '5' ) =>
+	setup( {
+		filterRating,
+		displayStyle: 'dropdown',
+		selectType: 'multiple',
+	} );
+
+describe( 'RatingFilterBlock', () => {
```

Comment:
I would try to use the 'official' block name here:
```suggestion
describe( 'Filter by Rating block', () => {
```
(This probably needs to be changed in the Filter by Attribute block tests as well)

Reply:
Sure thing! Changed in both: Filter by Rating and Attribute [in this commit](https://github.com/woocommerce/woocommerce-blocks/pull/7771/commits/df799fd4c89eb17271745d649ddafec0413a4740).